File: solutions/python/triangle/1/triangle.py
def equilateral(sides):
    if sides[0] == 0 or sides[1] == 0 and sides[2] == 0:
        return False 

    if (max(sides) == sides[0] and (sides[1] + sides[2]) < sides[0]) or \
       (max(sides) == sides[1] and (sides[0] + sides[2]) < sides[1]) or \
       (max(sides) == sides[2] and (sides[0] + sides[1]) < sides[2]):
        return False

    if sides[0] == sides[1] and sides[1] == sides[2]:
        return True
    return False


def isosceles(sides):
    if sides[0] == 0 or sides[1] == 0 and sides[2] == 0:
        return False 

    if (max(sides) == sides[0] and (sides[1] + sides[2]) < sides[0]) or \
       (max(sides) == sides[1] and (sides[0] + sides[2]) < sides[1]) or \
       (max(sides) == sides[2] and (sides[0] + sides[1]) < sides[2]):
        return False  

    if sides[0] == sides[1] or sides[1] == sides[2] or sides[0] == sides[2]:
        return True
    return False


def scalene(sides):
    if sides[0] == 0 or sides[1] == 0 and sides[2] == 0:
        return False  

    if (max(sides) == sides[0] and (sides[1] + sides[2]) < sides[0]) or \
       (max(sides) == sides[1] and (sides[0] + sides[2]) < sides[1]) or \
       (max(sides) == sides[2] and (sides[0] + sides[1]) < sides[2]):
        return False    
        
    if sides[0] != sides[1] and sides[1] != sides[2] and sides[0] != sides[2]:
        return True
    
    return False

File: solutions/go/triangle/1/triangle.go

// Package triangle determines what kind of triangle a triangle is
package triangle

import ("math"
		)

// Notice KindFromSides() returns this type. Pick a suitable data type.
type Kind int

const (
    // Pick values for the following identifiers used by the test program.
    NaT = iota // not a triangle
    Equ // equilateral
    Iso // isosceles
    Sca // scalene
)

// KindFromSides is a function that determines what kind of sides a triangle has.
func KindFromSides(a, b, c float64) Kind {

	if math.IsNaN(a) || math.IsNaN(b) || math.IsNaN(c) {
		return NaT
	}

	if math.IsInf(a,-1) || math.IsInf(b,-1) || math.IsInf(c,-1) ||
	   math.IsInf(a,1) || math.IsInf(b,1) || math.IsInf(c,1){
		return NaT
	}

	if a<=0.0 || b<=0.0 || c<= 0.0 {
		return NaT
	}

	if a+b < c || a+c <b || b+c < a {
		return NaT
	}


	if a==b && b == c {
		return Equ
	} else if a==b || a==c || b==c {
		return Iso
	}

	return Sca
}
